### PR TITLE
{partition-manager,libsForQt5.kpmcore}: 3.3.1 -> 4.2.0 and fix build

### LINKED ISSUE #76415: KDE Partition Manager not running with sudo
**Describe the bug**
If you open KDE Partition Manager using its ```.desktop``` file, it tells you that you aren't running it as root, and you won't be able to make any changes with it (rendering the program useless). The only workaround is running ```sudo partitionmanager``` (which is probably unintuitive to people with less linux experience) or changing the ```.desktop``` file (a bad idea when you want Nix to manage your system)

**To Reproduce**
1. Install ```partition-manager``` by adding it to your ```environment.systemPackages```
2. Run the ```.desktop``` file (e.g. in KDE, click on the start menu in the bottom left, type in partition manager, and click on the program)

**Expected behavior**
Use ```pkexec``` when clicking the ```.destop``` file for a seamless experience

**Screenshots**
image

**Additional context**
The message mentions ```kdesu``` so I tried adding it to ```environment.systemPackages``` by hand, but that did not work.

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.90, NixOS, 19.09.1682.bfdae0860e4 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(root): `"nixos-19.09.1682.bfdae0860e4"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### LINKED ISSUE #113942: `partition-manager` and `kpmcore` failed to build on master
**Describe the bug**
`partition-manager` and `kpmcore` failed to build on current master (92b1ef601c9a5032e877e09c52085978b2b56a9c).
(The former depends on the latter.)

Build log:
https://gist.github.com/oxalica/ca028f49640e911757dac37f474c4125

**To Reproduce**
1. Checkout nixpkgs master.
2. `nix build .\#partition-manager -L` (or  `nix build .\#libsForQt5.kpmcore -L`)
3. See the build error.

**Expected behavior**
It builds.

**Notify maintainers**

@peterhoeg 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.97, NixOS, 21.05.20210216.64b4617 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20201205_a5d85d0`
 - channels(root): `"nixos-21.03pre257339.83cbad92d73"`
 - channels(oxa): `"rust-overlay, home-manager, mozilla"`
 - nixpkgs: `/nix/store/fzx8mnbpbg2qjcac4958ni7xva0wsbyj-source`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - partition-manager
  - libsForQt5.kpmcore
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by oxalica**
###### Motivation for this change

Fix build failure and bump version of  `partition-manager` and `libsForQt5.kpmcore`.

`partition-manager` changed to polkit and droped the old kdesu impl.
It now requires polkit actions and dbus service, provided in `kpmcore`, to be installed.
So in this PR, a nixos module `programs.partition-manager.enable` is introduced. It should be enabled to make `partition-manager` work correctly. Note that password is still required every time launching it.

Fix #76415. Fix #113942.
Close #61077. (staled for a long time)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  - kpmcore: 294.1M -> 327.8M
  - partition-manager: 1.4G -> 1.5G
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114016` at 24ea8cfe run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/edd33c586636bb811733a30fd405bce2">calamares</a></li></li>
  </ul>
</details>

File: pkgs/development/libraries/kpmcore/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/kpmcore/default.nix:35]:
```suggestion
    util-linux # Needs blkid in configure script (note that this is not provided by util-linux-compat)
```

**Review by bachp — APPROVED** [review]
This makes kpmcore build again on master. Tested on Plasma.

File: pkgs/development/libraries/kpmcore/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/kpmcore/default.nix:1]:
```suggestion
{ stdenv, lib, fetchurl, fetchpatch, extra-cmake-modules
```

File: pkgs/tools/misc/partition-manager/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/misc/partition-manager/default.nix:62]: This could be clarified to gpl2Only or gpl2Plus.

File: pkgs/development/libraries/kpmcore/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/kpmcore/default.nix:42]: license, description and homepage are missing.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
otherwise the package looks good to me. Not sure about the module.

**Comment by billksun** [comment]
@SuperSandro2000, can you take a look please? Hoping this gets merged soon so my system can do updates again.

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000, can you take a look please? Hoping this gets merged soon so my system can do updates again.

I don't really review modules.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114016` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>libsForQt512.kpmcore</li>
</ul>
</details>
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>calamares</li>
<li>libsForQt5.kpmcore (libsForQt515.kpmcore ,plasma5Packages.kpmcore)</li>
<li>libsForQt514.kpmcore</li>
<li>partition-manager</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
calamares:
<details>


warning: build-tools-in-build-inputs
util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/misc/calamares/default.nix:18:3:

```
   |
18 |   buildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: unused-argument
Unused argument: `glibc`.
Near pkgs/tools/misc/calamares/default.nix:3:61:

```
  |
3 | , qtquickcontrols, qtsvg, qttools, qtwebengine, util-linux, glibc, tzdata
  |                                                             ^
```
</details>
</details>

**Comment by bachp** [comment]
Module looks OK to me.

**Comment by billksun** [comment]
> otherwise the package looks good to me. Not sure about the module.

@SuperSandro2000, you requested a change, so you must approve it before we can merge.

File: pkgs/development/libraries/kpmcore/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/kpmcore/default.nix:46]:
```suggestion
    maintainers = with maintainers; [ peterhoeg oxalica ];
```

File: pkgs/development/libraries/kpmcore/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/kpmcore/default.nix:48]:
```suggestion
    broken = versionOlder qtbase.version "5.14";
```

**Comment by oxalica** [comment]
@SuperSandro2000 
Fixed and squashed.

**Comment by billksun** [comment]
@SuperSandro2000, do the latest commits meet your change request?

**Review by SuperSandro2000 — APPROVED** [review]
Package LGTM, no idea if the module can be done like this.

**Pull request merged**
